RadixWallet/Features/AssetTransferFeature/AssetTransfer+Reducer.swift

```diff
@@ -232,30 +234,79 @@ extension AssetTransfer {
 
 func instructionForDepositing(
 	bucket: ManifestBuilderBucket,
+	resource: ResourceAddress,
 	into receivingAccount: ReceivingAccount.State.Account
 ) async throws -> ManifestBuilder.InstructionsChain.Instruction {
-	@Dependency(\.userDefaultsClient) var userDefaultsClient
-	@Dependency(\.secureStorageClient) var secureStorageClient
-	let isUserAccount = receivingAccount.isUserAccount
-	// TODO: Temporary revert of checking if the receiving account is a ledger account
-	let isSoftwareAccount = true // !receivingAccount.isLedgerAccount
 	let recipientAddress = receivingAccount.address
-	let userHasAccessToMnemonic = if let deviceFactorSourceID = receivingAccount.left?.deviceFactorSourceID {
-		await secureStorageClient.containsMnemonicIdentifiedByFactorSourceID(deviceFactorSourceID)
-	} else { false }
 
-	guard isUserAccount, isSoftwareAccount, userHasAccessToMnemonic else {
+	if case let .left(userAccount) = receivingAccount {
+		@Dependency(\.secureStorageClient) var secureStorageClient
+		// TODO: Temporary revert of checking if the receiving account is a ledger account
+		let isSoftwareAccount = true // !receivingAccount.isLedgerAccount
+		let userHasAccessToMnemonic = if let deviceFactorSourceID = userAccount.deviceFactorSourceID {
```

Comment:
I understand that we want to be modern and use Swift 5.9 but `= if let` hurts my eyes. Couldn't this just use `??`?